primap2/csg/_strategies/global_least_squares.py

```diff
@@ -0,0 +1,267 @@
+import numpy as np
+import xarray as xr
+from attrs import frozen
+from scipy.linalg import lstsq
+from scipy.optimize import least_squares
+
+import primap2
+
+
+@frozen
+class GlobalLSStrategy:
+    """Fill missing data by global least square matching.
+
+    The NaNs in the first timeseries `ts(t)` using harmonized data from the lower
+    priority timeseries `fill_ts(t)`. For harmonization we use
+    fill_ts(t)_h = ts(t) * a + b,
+    where fill_ts(t)_h is the harmonized dataset and a and b are determined by minimizing
+    the least squares distance between ts(t) and fill_ts(t)_h.
+
+    If the class is initialized with `allow_shift = True` the faster
+    `GlobalLSlstsqStrategy` which is based on the `scipy.linalg.lstsq` function
+    is used.
+    For the case `allow_shift = False` (b = 0) `scipi.optimize.least_squares` is used.
+
+    If there is no overlap in non-NaN data between ts(t) and fill_ts(t) the
+    `SubstitutionStrategy` is used.
+    """
+
+    allow_shift: bool = True
+    type = "globalLS"
+
+    def factor_mult(self, a, e, e_ref):
```

Comment:
```suggestion
    def _factor_mult(self, a, e, e_ref):
```
Really a very small thing, but if you don't want to make this function part of the public API of primap2, it should be marked private since the Strategy class will be part of the public API or primap2.